### DISCUSSION TITLE
tskit.load returns "unknown file attribute: i" on macOS

### DISCUSSION THREAD

**Discussion by chlangley** [tskit-dev/tsinfer#586] [Q&A] [answered]
Hello:

Just trying to work through the Example in the documentation.
When I get to the tskit.load():

source = tskit.load("simulation-source.trees")
inferred = tskit.load("simulation.trees")

I get this 
"zsh: unknown file attribute: i"

I trust I am ignorant of a simple python/tskit feature.  
Thanks for any help.

Cheers,
Chuck

**Comment by petrelharp** [comment]
Well, that's a confusing error. This is in [the tsinfer docs](https://tsinfer.readthedocs.io/en/latest/tutorial.html), so I'll transfer this over to tsinfer.

But, I've just tried out the code in the tutorial and not got the same problem. Maybe first make sure that tsinfer is up to date? And then check the file's in the directory your python is in, like:
```
os.path.isfile("simulation.trees")
```
Another check is to go to the terminal and check the `.trees` files are valid there:
```
tskit info simulation.trees
```

Let us know what happens?

**Reply by petrelharp** [reply]
(note: if there could be a more informative error message in tskit for whatever happened, that'd be a nice thing...)

**Reply by chlangley** [reply]
Here are the contents of the directory and the versions of tsinfer and tskit:

```
➜  playts ll
total 39172
-rw-r--r--@ 1 chuck  admin   641B Sep 24 03:23 simulation-example.py
-rw-r--r--  1 chuck  admin   8.2M Sep 24 03:25 simulation-source.trees
-rw-r--r--  1 chuck  admin    22M Sep 24 03:25 simulation.samples
-rw-r--r--  1 chuck  admin   7.8M Sep 25 10:19 simulation.trees
➜  playts conda activate
(base) ➜  playts tsinfer -V
tsinfer 0.2.1
(base) ➜  playts tskit -V
python3 -m tskit 0.3.7
```

**Reply by petrelharp** [reply]
That all seems in order - any ideas here @benjeffery @hyanwong?

**Reply by chlangley** [reply]
Does anyone think I should attempt a clean start, removing miniconda and all the ts modules?  

And I should have mentioned that my macOS is v11.6 and hardware is MacBook Pro (16 inch, 2019).

The python3 is v 3.9.7 and conda -V   4.10.3.

Thanks for any  help.
Cheers,  Chuck

**Reply by hyanwong** [reply]
It should be easy to install a new miniconda distro somewhere else and try that? I'm stumped here: I've never seen that before and the internet is surprisingly unhelpful on this one.

**Reply by petrelharp** [reply]
Any other diagnostics you can think of? @chlangley what happens when you run `tskit info simulation.trees` and `tskit info simulation-source.trees`? (Also, which .trees file gives the error?)

**Reply by chlangley** [reply]
(base) ➜  playts ll
total 38888
-rw-r--r--@ 1 chuck  admin   641B Sep 24 03:23 simulation-example.py
-rw-r--r--  1 chuck  admin   8.2M Sep 24 03:25 simulation-source.trees
-rw-r--r--  1 chuck  admin    22M Sep 24 03:25 simulation.samples
-rw-r--r--  1 chuck  admin   7.8M Sep 25 10:19 simulation.trees
(base) ➜  playts source = tskit.load("simulation-source.trees")
zsh: unknown file attribute: i
(base) ➜  playts inferred = tskit.load("simulation.trees")
zsh: unknown file attribute: i
(base) ➜  playts tskit info simulation.trees
sequence_length:  10000000.0
trees:            19838
samples:          10000
individuals:      10000
nodes:            37776
edges:            98601
sites:            38778
mutations:        38778
migrations:       0
populations:      0
provenances:      4
(base) ➜  playts

**Comment by petrelharp** [comment]
Ah-ha! I *think* the answer is simple - when I do that *in zsh* I get:
```
$ source = tskit.load("simulation-source.trees")
zsh: unknown file attribute
```
... which is almost the same error.  However - the code chunk starting
```
import tskit

source = tskit.load("simulation-source.trees")
inferred = tskit.load("simulation.trees")
```
is supposed to be run in python, not the shell. Is this the problem? We haven't said explicitly that we're switching back to python here, I guess that we're expecting the user to pick up on the lack of `$`s at the start of the lines? Perhaps we should say "Going back to python," before that code chunk?

**Reply by chlangley** [reply]
yes.  if I 'conda deactivate'.  And then invoke python3, then I 

import tskit

source = tskit.load("simulation-source.trees")
inferred = tskit.load("simulation.trees")

and all seems good.  

So these tskit command can not be called with conda?

**Reply by benjeffery** [reply]
Note that `conda activate` does not start python, it only sets which python is avaliable. You need to run `python3` to run any python code.

**Reply by hyanwong** [reply]
We can probably tidy the language in the tutorial to make it clearer. I'm happy to do this.

**Reply by hyanwong** [reply]
Can you mark this as the correct answer @chlangley ?

**Comment by hyanwong** [comment]
I've changed the tutorial wording @chlangley - could you see if it's clearer now that this needs to be run within python, rather than on the command line?

**Comment by chlangley** [comment, marked as answer]
Will do
